packer/windows/scripts/install-docker.ps1

```diff
@@ -1,7 +1,7 @@
 # Stop script execution when a non-terminating error occurs
 $ErrorActionPreference = "Stop"
 
-$docker_version="20.10.9"
+$docker_version="20.10.23"
```

Comment:
Looks like the latest version from the DockerMsftProvider is still
20.10.9. The build logs have:
```
[0;32m2023-02-22T01:10:00Z:     amazon-ebs: Upgrading DockerMsftProvider module[0m

[0;32m2023-02-22T01:10:09Z:     amazon-ebs: Printing available docker versions[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs:[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Name                           Version          Source           Summary[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: ----                           -------          ------           -------[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         17.06.2-ee-25    DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         18.03.1-ee-12    DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         18.09.14         DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         19.03.19         DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         20.10.9          DockerDefault    Contains Docker EE for use with Windows Server.[0m
```

Reply:
This is the cause of the build failure on [`f87c3da` (#1093)](https://github.com/buildkite/elastic-ci-stack-for-aws/pull/1093/commits/f87c3da08f5a312d2649d127c356c9feb445c406)